cosmoz-treenode-navigator.js

```diff
@@ -162,108 +163,49 @@
-						return {
-							id: child.key,
-							name: child[this.comparisonProperty],
-							path: child.pathLocator,
-							children: child[this.childProperty]
-						};
-					}, this);
-				} else {
-					pathSegment = children;
-				}
-			}, this);
-
-			return this._sortItOut(level);
+			var n = tree.getNodeByPathLocator(pathLocator),
+				level = tree.getChildren(n) ? tree.getChildren(n) : n;
+			return this._sortNodes(this._normalizeNode(level));
 		},
 		/**
-		 * Returns an Array of nodes on a given path.
+		 * Returns the "formatted" normalized node
+		 * with the properties id, name, path, sectionName, children
 		 */
-		_getNodesOnPath: function (pl, nodes) {
-			var path = pl.split(this.separatorSign),
-				pathSegment = nodes;
-				
-			return path.map(function (nodeKey) {
-				var node = pathSegment[nodeKey],
-					children = node[this.childProperty];
-				node['key'] = nodeKey;
-				if (node && children !== undefined && Object.keys(children).length > 0) {
-					pathSegment = children;
-				}
-				return node;
+		_normalizeNode: function (nodeObj) {
+			if (!nodeObj) {
+				return;
+			}
+			return Object.keys(nodeObj).map(function (key) {
+				var node = nodeObj[key];
+				return {
+					id: key,
+					name: node[this.comparisonProperty],
+					path: node.pathLocator || node.path,
+					sectionName: this.tree.getPathString(node.path, this.comparisonProperty),
+					children: this.tree.getChildren(node)
+				};
 			}, this);
 		},
 		/**
 		 * Returns a node based on a given path locator.
+		 * If pathLocator is empty or not defined, null gets returned.
 		 */
-		_getNode: function (pl, nodes) {
-			if (!pl) {
+		_getNode: function (pathLocator) {
+			if (!pathLocator) {
 				return null;
 			}
-			var pathArray = pl.split(this.separatorSign),
-				pathSegment = nodes,
-				node;
-			
-			pathArray.some(function (path) {
-				node = pathSegment[path];
-
-				if (node === undefined) {
-					console.error('Path does not exist.', pathArray, path, nodes);
-					return true;
-				}
-				var children = node[this.childProperty];
-				if (children !== undefined && Object.keys(children).length > 0) {
-					pathSegment = children;
-				}
-			}, this);
-
-			return node;
+			var node = this.tree.getNodeByPathLocator(pathLocator);
+			return this._normalizeNode(node);
 		},
-		_getTreePathParts: function (value, data) {
-			if (value === null) {
-				value = '';
-			}
-			var path = value.split(this.separatorSign),
-				parts = [],
-				newpath = [];
-
-			path.some(function (part) {
-				newpath.push(part);
-				var newPathString = newpath.join(this.separatorSign),
-					node = this._getNode(newPathString, data);
-				if (!node) {
-					return true;
-				}
-				node.path = newPathString;
-				parts.push(node);
-			}, this);
-
-			return parts;
+		_getTreePathParts: function (pathLocator, tree) {
+			if (!tree || !pathLocator) {
+				return [];
+			};
+			return this._normalizeNode(tree.getPath(pathLocator));
```

Comment:
`getPath()` returns an array, need `_normalizeNodes()`
But I don't think we need to normalize these, do we ?